docs/csharp/language-reference/compiler-messages/pattern-matching-warnings.md

```diff
@@ -0,0 +1,20 @@
+---
+title: Pattern matching warnings
+description: There are several pattern matching warnings. Learn how to address these warnings.
+f1_keywords:
+  - "CS8509" # WRN_SwitchNotAllPossibleValues: The switch expression does not handle all possible values of its input type (it is not exhaustive). For example, the pattern '...' is not covered.
+helpviewer_keywords:
+  - "CS8509"
+ms.date: 11/10/2022
+---
+# Pattern matching warnings
+
+There are several pattern matching warnings. Learn how to address these warnings.
+
+- **CS8509** - *The switch expression does not handle all possible values of its input type (it is not exhaustive). For example, the pattern '...' is not covered.*
+
+The following code snippets generate CS8509:
+
+:::code language="csharp" source="./snippets/pattern-matching-warnings/Switch.cs" id="SwitchNotAllPossibleValues":::
+
+For more information, see [Switch](../statements/selection-statements.md#the-switch-statement).
```

Comment:
Let's add a paragraph about how a reader can address this warning:

```suggestion
To address this warning, add switch arms that cover all possible input values. For example:

```csharp
var result = enumValues switch
{
    EnumValues.Value1 => 1,
    EnumValues.Value2 => 2,
    EnumValues.Value3 => 3,
    _ => throw new ArgumentException("Input isn't a valid enum value", nameof(enumValues));
};
````

The `_` pattern matches all remaining values. It's often used to match invalid values, as shown in the preceding example.

For more information, see [Switch](../statements/selection-statements.md#the-switch-statement).
```

Reply:
The last sentences of the previous comment should be added / restored as well. The first set of backticks is ending the entire suggestion, where I meant just to end the C# sample.

Also, it would be better to update the code file by adding a new method, but that's a lot harder to do as a suggestion in the UI.